go-controller/pkg/ovn/gateway.go

```diff
@@ -760,3 +763,167 @@ func (gw *GatewayManager) addExternalSwitch(prefix, interfaceID, nodeName, gatew
 
 	return nil
 }
+
+// Cleanup removes all the NB DB objects created for a node's gateway
+func (gw *GatewayManager) Cleanup() error {
+	// Get the gateway router port's IP address (connected to join switch)
+	var nextHops []net.IP
+
+	gwIPAddrs, err := libovsdbutil.GetLRPAddrs(gw.nbClient, types.GWRouterToJoinSwitchPrefix+gw.gwRouterName)
+	if err != nil && !errors.Is(err, libovsdbclient.ErrNotFound) {
+		return err
+	}
+
+	for _, gwIPAddr := range gwIPAddrs {
+		nextHops = append(nextHops, gwIPAddr.IP)
+	}
+	gw.staticRouteCleanup(nextHops)
+	gw.policyRouteCleanup(nextHops)
+
+	// Remove the patch port that connects join switch to gateway router
+	portName := types.JoinSwitchToGWRouterPrefix + gw.gwRouterName
+	lsp := nbdb.LogicalSwitchPort{Name: portName}
+	sw := nbdb.LogicalSwitch{Name: gw.transitSwitchName}
```

Comment:
nit; this is an IC only switch and you dont check if youre in IC mode or not. Doesnt matter i guess since `DeleteLogicalSwitchPorts` wont error if not found.